slatedb/src/batch_write.rs

```diff
@@ -207,19 +207,17 @@ impl DbInner {
         Some(spawn_bg_task(
             tokio_handle,
             move |result| {
-                let err = match result {
+                match result {
                     Ok(()) => {
                         info!("write task shutdown complete");
-                        SlateDBError::BackgroundTaskShutdown
                     }
                     Err(err) => {
                         warn!("write task exited [error={}]", err);
-                        err.clone()
+                        // notify any waiters about the failure
+                        let mut state = this.state.write();
+                        state.record_fatal_error(err.clone());
                     }
-                };
-                // notify any waiters about the failure
-                let mut state = this.state.write();
-                state.record_fatal_error(err.clone());
```

Comment:
am I reading this correct that before this change we used to alert the waiters about the `SlateDBError:BackgroundTaskShutdown` (even the "clean" path) but after this patch we don't? we should make sure we aren't losing anything in translation - when I was looking around the codebase it looks like there are quite a few places the stat error watcher write-once cell is used.

Reply:
I have added a drain_write_messages() function that sends SlateDBError::Shutdown to any pending write requests before the task exits. Now waiters get proper notification instead of hanging. 

can you please review and let me know if this helps or not?